pkg/operator/starter.go

```diff
@@ -481,6 +483,24 @@ func hyperShiftReplaceNamespaceHook(operandNamespace string) dc.DeploymentHookFu
 	}
 }
 
+// hyperShiftSetSecurityContext sets the RunAsNonRoot to false and sets the RunAsUser to 1001 when the management
+// cluster does not support SCC. This was added to allow HyperShift to run on Azure Kubernetes Service (AKS) as its
+// management cluster for Azure Red Hat OpenShift (ARO) HCP. A patch was added to the HyperShift repo to set this OS
+// environment variable if the management cluster HyperShift is running supports SCC.
+//
+// See https://issues.redhat.com/browse/STOR-1805 for further details.
+func hyperShiftSetSecurityContext() dc.DeploymentHookFunc {
+	return func(_ *operatorv1.OperatorSpec, deployment *appsv1.Deployment) error {
+		hyperShiftSupportsSCC, ok := os.LookupEnv("HYPERSHIFT_SUPPORTS_SCC")
```

Comment:
cluster-network-operator already uses env. var. [RUN_AS_USER](https://github.com/jsafrane/hypershift/blob/5e3ec5d84af183a798c06662313ca87835d85ad3/control-plane-operator/controllers/hostedcontrolplane/cno/clusternetworkoperator.go#L361), I think it would be better to use the same (and its value) to be consistent.